--- kbd-1.05/man/man8/kbdrate.8.jj	Tue Apr 17 11:07:57 2001
+++ kbd-1.05/man/man8/kbdrate.8	Tue Apr 17 11:08:24 2001
@@ -22,7 +22,7 @@
 without any options will reset the repeat rate to 10.9 characters per second (cps)
 and the delay to 250 milliseconds (ms) for Intel- and M68K-based systems.
 These are the IBM defaults. On SPARC-based systems it will reset the repeat rate
-to 5 cps and the delay to 200 ms.
+to 20 cps and the delay to 200 ms.
 
 .SH OPTIONS
 .TP
@@ -66,3 +66,5 @@
 .I /etc/rc.local
 .br
 .I /dev/port
+.br
+.I /dev/kbd
--- kbd-1.05/src/kbdrate.c.jj	Tue Apr 17 11:06:19 2001
+++ kbd-1.05/src/kbdrate.c	Tue Apr 17 11:07:30 2001
@@ -98,8 +98,11 @@
 
 static int
 KDKBDREP_ioctl_ok(double rate, int delay, int silent) {
+#ifndef __sparc__
 	/* This ioctl is defined in <linux/kd.h> but is not
-	   implemented anywhere - must be in some m68k patches. */
+	   implemented anywhere - must be in some m68k patches.
+	   Cannot blindly try unimplemented ioctls on sparc64 -
+	   the 32<->64bit transition layer does not like it. */
 	struct kbd_repeat kbdrep_s;
 
 	/* don't change, just test */
@@ -153,6 +155,10 @@
 			rate, kbdrep_s.delay );
 
 	return 1;			/* success! */
+
+#else
+	return 0;
+#endif /* __sparc__ */
 }

 static int
@@ -182,7 +184,7 @@
 int main( int argc, char **argv )
 {
 #ifdef __sparc__
-	double      rate = 5.0;      /* Default rate */
+	double      rate = 20.0;      /* Default rate */
 	int         delay = 200;     /* Default delay */
 #else
 	double      rate = 10.9;     /* Default rate */
@@ -230,8 +232,9 @@
 		return 0;
 
 
-	/* The ioport way */
+	/* The ioport way - will crash on sparc */
 
+#ifndef __sparc__
 	for (i = 0; i < RATE_COUNT; i++)
 		if (rate * 10 >= valid_rates[i]) {
 			value &= 0x60;
@@ -276,6 +279,8 @@
 		printf( _("Typematic Rate set to %.1f cps (delay = %d ms)\n"),
 			valid_rates[value & 0x1f] / 10.0,
 			valid_delays[ (value & 0x60) >> 5 ] );
+
+#endif
 
 	return 0;
 }
