--- kbd-1.06/src/unicode_start.orig	Mon Feb 12 17:30:17 2001
+++ kbd-1.06/src/unicode_start	Sat Jun  2 16:29:26 2001
@@ -18,7 +18,13 @@
 # produce UTF-8 encoded multibyte sequences, instead of single bytes
 # >= 0x80 in a legacy 8-bit encoding.
 
-dumpkeys | loadkeys --unicode
+# PLD: disabled because it breaks non-iso-8859-1 keymaps
+# instead use manually:
+#   loadkeys --unicode <keymap>
+# or
+#   dumpkeys -c <charset> | loadkeys --unicode
+
+#dumpkeys | loadkeys --unicode
 
 # 2. The output side: the console screen.
 
@@ -32,7 +38,9 @@
 # have a Unicode map attached, or explicitly specified, e.g.,
 # by giving `def.uni' as a second argument.
 
-DEFAULT_UNICODE_FONT=LatArCyrHeb-16
+# note: no need to reload font if one with good unicode map is loaded already.
+
+#DEFAULT_UNICODE_FONT=LatArCyrHeb-16
 # Also drdos8x16 is a good candidate.
 
 case $# in
@@ -43,7 +51,9 @@
 		setfont $1
 		;;
 	0)
-		setfont $DEFAULT_UNICODE_FONT
+		if [ -n "$DEFAULT_UNICODE_FONT" ]; then
+			setfont $DEFAULT_UNICODE_FONT
+		fi
 		;;
 	*)
 		echo "usage: unicode_start [font [unicode map]]"
