--- kbd-1.06/src/loadkeys.y.orig	Fri May 18 16:29:05 2001
+++ kbd-1.06/src/loadkeys.y	Sat Jun  2 16:13:18 2001
@@ -596,7 +596,7 @@
 }
 
 char *dirpath[] = { "", DATADIR "/" KEYMAPDIR "/**", KERNDIR "/", 0 };
-char *suffixes[] = { "", ".map", 0 };
+char *suffixes[] = { "", ".map", ".kmap", 0 };
 extern FILE *findfile(char *fnam, char **dirpath, char **suffixes);
 
 #undef yywrap
--- kbd-1.06/src/loadunimap.c.orig	Mon Feb 12 23:55:39 2001
+++ kbd-1.06/src/loadunimap.c	Sat Jun  2 16:14:53 2001
@@ -28,7 +28,7 @@
 extern int force;
 
 static char *unidirpath[] = { "", DATADIR "/" UNIMAPDIR "/", 0 };
-static char *unisuffixes[] = { "", ".uni", 0 };
+static char *unisuffixes[] = { "", ".uni", ".sfm", 0 };
 
 #ifdef MAIN
 #include "version.h"
--- kbd-1.06/src/mapscrn.c.orig	Mon Feb 12 18:26:05 2001
+++ kbd-1.06/src/mapscrn.c	Sat Jun  2 16:16:23 2001
@@ -24,7 +24,7 @@
 
 /* search for the map file in these directories (with trailing /) */
 static char *mapdirpath[] = { "", DATADIR "/" TRANSDIR "/", 0 };
-static char *mapsuffixes[] = { "", ".trans", "_to_uni.trans", 0 };
+static char *mapsuffixes[] = { "", ".trans", "_to_uni.trans", ".acm", 0 };
 
 #ifdef MAIN
 #include "getfd.h"
