--- kbd-1.08/configure~	2003-07-04 14:17:45.000000000 +0000
+++ kbd-1.08/configure	2003-07-04 14:19:13.000000000 +0000
@@ -151,14 +151,14 @@
 #
 # 2. For lib/nls.h: do we have <libintl.h> and gettext() ?
 #
-echo '
+cat >conftest.c<<_ACEOF
 #include <libintl.h>
 main(int a, char **v){
   if (a == -1)  /* false */
         gettext("There is no gettext man page\n");
   exit(0);
 }
-' > conftest.c
+_ACEOF
 eval $compile
 if [ $nls = 1 ]; then
 	if test -s conftest && ./conftest 2>/dev/null; then
