--- kbd-1.15/configure.ac.orig	2008-11-28 22:10:15.000000000 +0100
+++ kbd-1.15/configure.ac	2008-12-05 20:28:32.692289465 +0100
@@ -75,7 +75,7 @@
 # Does xgettext exist and take the option --foreign-user?
 FOREIGN=
 HAVE_XGETTEXT=no
-if (test "$enable_nls" = "yes" && type xgettext > /dev/null 2>&1); then
+if (test "$enable_nls" = "yes" && test -x /usr/bin/xgettext > /dev/null 2>&1); then
 	msg=`xgettext --foreign-user 2>&1 | grep unrecognized`
 	if test -z "$msg"; then
 		FOREIGN="--foreign-user"
