diff -urN kbd-1.09.org/Makefile.in kbd-1.09/Makefile.in
--- kbd-1.09.org/Makefile.in	2004-01-03 12:38:37.304752523 +0100
+++ kbd-1.09/Makefile.in	2004-01-03 12:38:43.603442541 +0100
@@ -1,10 +1,9 @@
 # Note: BINDIR, DATA_DIR, and DESTDIR also occur in src/Makefile.
-DESTDIR = @prefix@
 DATA_DIR = @datadir@
 MAN_DIR = @mandir@
-BINDIR  = $(DESTDIR)/bin
-MANDIR  = $(MAN_DIR)
-DATADIR = $(DATA_DIR)
+BINDIR  = $(DESTDIR)/usr/bin
+MANDIR  = $(DESTDIR)$(MAN_DIR)
+DATADIR = $(DESTDIR)$(DATA_DIR)
 # If you change the names of any of the following subdirs,
 # also change paths.h.
 OLDKEYMAPDIR = keytables
diff -urN kbd-1.09.org/po/Makefile kbd-1.09/po/Makefile
--- kbd-1.09.org/po/Makefile	2004-01-03 12:38:38.120582849 +0100
+++ kbd-1.09/po/Makefile	2004-01-03 12:39:02.391535035 +0100
@@ -14,7 +14,7 @@
 FOREIGN=
 SHELL = /bin/sh
 
-prefix = ${DESTDIR}
+prefix = ${DESTDIR}/usr
 exec_prefix = ${prefix}
 datadir = $(prefix)/share
 localedir = $(datadir)/locale
