--- kbd-2.5.1/src/unicode_stop.orig	2022-08-20 19:58:25.772620769 +0200
+++ kbd-2.5.1/src/unicode_stop	2022-08-20 19:59:19.465663223 +0200
@@ -1,5 +1,7 @@
 #!/bin/sh
 
+[ -x /usr/bin/tty ] || exit 0
+
 TTY=$(/usr/bin/tty)
 case "$TTY" in
 	/dev/console|/dev/vc*|/dev/tty[0-9]*)
--- kbd-2.5.1/src/unicode_start.orig	2022-08-20 19:58:25.775954085 +0200
+++ kbd-2.5.1/src/unicode_start	2022-08-20 19:59:38.195561754 +0200
@@ -1,5 +1,7 @@
 #!/bin/sh
 
+[ -x /usr/bin/tty ] || exit 0
+
 # 0. Check whether we're on a console
 TTY=$(/usr/bin/tty)
 case "$TTY" in
