diff -urN aaa/kbd-1.08/Makefile.in kbd-1.08/Makefile.in
--- aaa/kbd-1.08/Makefile.in	Sun Oct 13 01:11:52 2002
+++ kbd-1.08/Makefile.in	Sat Nov  2 17:31:44 2002
@@ -1,10 +1,9 @@
 # Note: BINDIR, DATA_DIR, and DESTDIR also occur in src/Makefile.
-DESTDIR = @prefix@
 DATA_DIR = @datadir@
 MAN_DIR = @mandir@
-BINDIR  = $(DESTDIR)/bin
-MANDIR  = $(MAN_DIR)
-DATADIR = $(DATA_DIR)
+BINDIR  = $(DESTDIR)/usr/bin
+MANDIR  = $(DESTDIR)$(MAN_DIR)
+DATADIR = $(DESTDIR)$(DATA_DIR)
 # If you change the names of any of the following subdirs,
 # also change paths.h.
 OLDKEYMAPDIR = keytables
diff -urN aaa/kbd-1.08/po/Makefile kbd-1.08/po/Makefile
--- aaa/kbd-1.08/po/Makefile	Sun Oct 13 01:20:50 2002
+++ kbd-1.08/po/Makefile	Sat Nov  2 17:38:49 2002
@@ -14,7 +14,7 @@
 FOREIGN=
 SHELL = /bin/sh
 
-prefix = ${DESTDIR}
+prefix = ${DESTDIR}/usr
 datadir = $(prefix)/share
 localedir = $(datadir)/locale
 gnulocaledir = $(prefix)/share/locale
