diff -Nur kbd-1.09.orig/openvt/openvt.c kbd-1.09/openvt/openvt.c
--- kbd-1.09.orig/openvt/openvt.c	2004-01-02 04:42:27.000000000 +0100
+++ kbd-1.09/openvt/openvt.c	2004-01-03 16:07:57.933400296 +0100
@@ -53,6 +53,10 @@
    char vtname[sizeof VTNAME + 2]; /* allow 999 possible VTs */
    char *cmd = NULL, *def_cmd = NULL, *username = NULL;
 
+   setlocale(LC_ALL, "");
+   bindtextdomain(PACKAGE, LOCALEDIR);
+   textdomain(PACKAGE);
+
    /*
     * I don't like using getopt for this, but otherwise this gets messy.
     * POSIX/Gnu getopt forces the use of -- to separate child/program
@@ -240,7 +244,7 @@
 
 	 if (ioctl(fd, VT_WAITACTIVE, vtno)){
             int errsv = errno;
-	    fprintf(stderr, "\nopenvt: activation interrupted? (%s)\n",
+	    fprintf(stderr, _("\nopenvt: activation interrupted? (%s)\n"),
 		    strerror(errsv));
 	    fflush(stderr);
 	    _exit (1);
@@ -264,7 +268,7 @@
    }
 
    if ( pid < 0 ) {
-      perror("openvt: fork() error");
+      perror(_("openvt: fork() error"));
       return(6);
    }
 
@@ -295,7 +299,7 @@
 void usage(int stat)
 {
    fprintf(stderr,
-     "Usage: openvt [-c vtnumber] [-l] [-u] [-s] [-v] [-w] -- command_line\n");
+     _("Usage: openvt [-c vtnumber] [-l] [-u] [-s] [-v] [-w] -- command_line\n"));
    exit (stat);
 }
 
@@ -360,7 +364,7 @@
 
 	/* get the owner of current tty */
 	if (!(pwnam = getpwuid(console_uid))) {
-		perror("can't getpwuid");
+		perror(_("can't getpwuid"));
 		exit(1);
 	}
 
@@ -375,7 +379,7 @@
 			goto got_a_process;
 	}
 
-	fprintf(stderr,"couldn't find owner of current tty!\n");
+	fprintf(stderr,_("couldn't find owner of current tty!\n"));
 	exit(1);
 
    got_a_process:
diff -Nur kbd-1.09.orig/po/POTFILES.in kbd-1.09/po/POTFILES.in
--- kbd-1.09.orig/po/POTFILES.in	2002-12-06 16:49:48.000000000 +0100
+++ kbd-1.09/po/POTFILES.in	2004-01-03 16:08:34.474845152 +0100
@@ -5,6 +5,7 @@
 openvt/getfd.c
 openvt/openvt.c
 po/cat-id-tbl.c
+src/analyze.c
 src/chvt.c
 src/clrunimap.c
 src/deallocvt.c
@@ -18,6 +19,7 @@
 src/kbdrate.c
 src/kdfontop.c
 src/ksyms.c
+src/loadkeys.c
 src/loadunimap.c
 src/mapscrn.c
 src/outpsfheader.c
diff -Nur kbd-1.09.orig/src/analyze.l kbd-1.09/src/analyze.l
--- kbd-1.09.orig/src/analyze.l	1999-10-06 04:52:48.000000000 +0200
+++ kbd-1.09/src/analyze.l	2004-01-03 16:09:07.820775800 +0100
@@ -67,7 +67,7 @@
 			 BEGIN(0);
 			}
 <INCLSTR>[^"]|\"\"|\"[^"\n]*{Eol}	{
-			  yyerror("expected filename between quotes");
+			  yyerror(_("expected filename between quotes"));
 			  BEGIN(0); }
 {Continuation}		{line_nr++;}
 {Eol}			{line_nr++;BEGIN(0);return(EOL);}
@@ -121,5 +121,5 @@
 
 void
 stringovfl(void) {
-	lkfatal("string too long");
+	lkfatal(_("string too long"));
 }
diff -Nur kbd-1.09.orig/src/loadkeys.y kbd-1.09/src/loadkeys.y
--- kbd-1.09.orig/src/loadkeys.y	2004-01-02 02:59:58.000000000 +0100
+++ kbd-1.09/src/loadkeys.y	2004-01-03 16:11:21.032524536 +0100
@@ -250,12 +250,12 @@
 "valid options are:\n"
 "\n"
 "	-c --clearcompose clear kernel compose table\n"
-"	-d --default	  load \"" DEFMAP "\"\n"
+"	-d --default	  load \"%s\"\n"
 "	-h --help	  display this help text\n"
 "	-m --mktable      output a \"defkeymap.c\" to stdout\n"
 "	-s --clearstrings clear kernel string table\n"
 "	-u --unicode      implicit conversion to Unicode\n"
-"	-v --verbose      report the changes\n"), VERSION);
+"	-v --verbose      report the changes\n"), VERSION, DEFMAP);
 	exit(1);
 }
 
@@ -286,6 +286,10 @@
 
 	set_progname(argv[0]);
 
+	setlocale(LC_ALL, "");
+	bindtextdomain(PACKAGE, LOCALEDIR);
+	textdomain(PACKAGE);
+
 	while ((c = getopt_long(argc, argv,
 		short_opts, long_opts, NULL)) != -1) {
 		switch (c) {
--- kbd-1.10/po/pl.po.orig	2004-01-03 16:03:41.000000000 +0100
+++ kbd-1.10/po/pl.po	2004-01-03 16:31:06.730271080 +0100
@@ -78,11 +79,46 @@
 "\n"
 "openvt: nie mo¿na otworzyæ %s w trybie R/W (%s)\n"
 
-#: openvt/openvt.c:291
+#: openvt/openvt.c:247
+#, c-format
+msgid ""
+"\n"
+"openvt: activation interrupted? (%s)\n"
+msgstr ""
+"\n"
+"openvt: uaktywnianie przerwane? (%s)\n"
+
+#: openvt/openvt.c:271
+msgid "openvt: fork() error"
+msgstr "openvt: b³±d fork()"
+
+#: openvt/openvt.c:295
 #, c-format
 msgid "openvt: could not deallocate console %d\n"
 msgstr "openvt: nie mo¿na zwolniæ konsoli %d\n"
 
+#: openvt/openvt.c:302
+#, c-format
+msgid "Usage: openvt [-c vtnumber] [-l] [-u] [-s] [-v] [-w] -- command_line\n"
+msgstr "Sk³adnia: openvt [-c numervt] [-l] [-u] [-s] [-v] [-w] -- linia_poleceñ\n"
+
+#: openvt/openvt.c:367
+msgid "can't getpwuid"
+msgstr "nie mo¿na wykonaæ getpwuid"
+
+#: openvt/openvt.c:382
+#, c-format
+msgid "couldn't find owner of current tty!\n"
+msgstr "nie mo¿na odnale¼æ w³a¶ciciela aktualnego tty!\n"
+
+#: analyze.l:70
+msgid "expected filename between quotes"
+msgstr "oczekiwano nazwy pliku w cudzys³owach"
+
+#: analyze.l:125
+msgid "string too long"
+msgstr "³añcuch za d³ugi"
+
 #: src/chvt.c:28
 #, c-format
 msgid "usage: chvt N\n"
@@ -428,6 +464,246 @@
 msgid "plus before %s ignored\n"
 msgstr "zignorowano plus przed %s\n"
 
+#: loadkeys.y:149
+#, c-format
+msgid "'%s' is not a function key symbol"
+msgstr "'%s' nie jest symbolem klawisza funkcyjnego"
+
+#: loadkeys.y:212
+#, c-format
+msgid "too many (%d) entries on one line"
+msgstr "za du¿o (%d) wpisów w jednej linii"
+
+#: loadkeys.y:225
+msgid "too many keydefinitions on one line"
+msgstr "za du¿o definicji klawiszy w jednej linii"
+
+#: loadkeys.y:246
+#, c-format
+msgid ""
+"loadkeys version %s\n"
+"\n"
+"Usage: loadkeys [option...] [mapfile...]\n"
+"\n"
+"valid options are:\n"
+"\n"
+"\t-c --clearcompose clear kernel compose table\n"
+"\t-d --default\t  load \"%s\"\n"
+"\t-h --help\t  display this help text\n"
+"\t-m --mktable      output a \"defkeymap.c\" to stdout\n"
+"\t-s --clearstrings clear kernel string table\n"
+"\t-u --unicode      implicit conversion to Unicode\n"
+"\t-v --verbose      report the changes\n"
+msgstr ""
+"loadkeys wersja %s\n"
+"\n"
+"Sk³adnia: loadkeys [opcja...] [plik-mapy...]\n"
+"\n"
+"poprawne opcje to:\n"
+"\n"
+"\t-c --clearcompose wyczyszczenie tablicy compose w j±drze\n"
+"\t-d --default\t  wczytanie \"%s\"\n"
+"\t-h --help\t  wy¶wietlenie tego opisu\n"
+"\t-m --mktable      wypisanie \"defkeymap.c\" na standardowe wyj¶cie\n"
+"\t-s --clearstrings wyczyszczenie tablicy ³añcuchów w j±drze\n"
+"\t-u --unicode      niejawna konwersja do Unikodu\n"
+"\t-v --verbose      raportowanie zmian\n"
+
+#: loadkeys.y:329
+#, c-format
+msgid "syntax error in map file\n"
+msgstr "b³±d sk³adni w pliku mapy\n"
+
+#: loadkeys.y:331
+#, c-format
+msgid "key bindings not changed\n"
+msgstr "przypisania klawiszy nie zmienione\n"
+
+#: loadkeys.y:388
+msgid "includes nested too deeply"
+msgstr "do³±czane pliki zbyt g³êboko zagnie¿d¿one"
+
+#. start reading include file
+#: loadkeys.y:544
+#, c-format
+msgid "switching to %s\n"
+msgstr "prze³±czanie na %s\n"
+
+#: loadkeys.y:550
+#, c-format
+msgid "cannot open include file %s"
+msgstr "nie mo¿na otworzyæ do³±czanego pliku %s"
+
+#: loadkeys.y:597
+#, c-format
+msgid "Cannot find %s\n"
+msgstr "Nie mo¿na znale¼æ %s\n"
+
+#: loadkeys.y:610
+#, c-format
+msgid "cannot open file %s\n"
+msgstr "nie mo¿na otworzyæ pliku %s\n"
+
+#: loadkeys.y:621
+#, c-format
+msgid "Loading %s\n"
+msgstr "Wczytywanie %s\n"
+
+#: loadkeys.y:633
+#, c-format
+msgid "addmap called with bad index %d"
+msgstr "addmap wywo³ane z b³êdnym indeksem %d"
+
+#: loadkeys.y:637
+#, c-format
+msgid "adding map %d violates explicit keymaps line"
+msgstr "dodanie mapy %d narusza bezpo¶rednio podan± liniê keymaps"
+
+#: loadkeys.y:651
+#, c-format
+msgid "killkey called with bad index %d"
+msgstr "killkey wywo³ane z b³êdnym indeksem %d"
+
+#: loadkeys.y:653
+#, c-format
+msgid "killkey called with bad table %d"
+msgstr "killkey wywo³ane z b³êdn± tablic± %d"
+
+#: loadkeys.y:667
+#, c-format
+msgid "addkey called with bad index %d"
+msgstr "addkey wywo³ane z b³êdnym indeksem %d"
+
+#: loadkeys.y:669
+#, c-format
+msgid "addkey called with bad table %d"
+msgstr "addkey wywo³ane z b³êdn± tablic± %d"
+
+#: loadkeys.y:711
+#, c-format
+msgid "%s: addfunc called with bad func %d\n"
+msgstr "%s: addfunc wywo³ane z b³êdn± funkcj± %d\n"
+
+#: loadkeys.y:739
+#, c-format
+msgid "%s: addfunc: func_buf overflow\n"
+msgstr "%s: addfunc: przepe³nienie func_buf\n"
+
+#: loadkeys.y:757
+#, c-format
+msgid "compose table overflow\n"
+msgstr "przepe³nienie tablicy compose\n"
+
+#: loadkeys.y:794
+#, c-format
+msgid "Keymap %d: Permission denied\n"
+msgstr "Mapa %d: Brak uprawnieñ\n"
+
+#: loadkeys.y:803
+msgid "    FAILED"
+msgstr "    NIE POWIOD£O SIÊ"
+
+#: loadkeys.y:806
+#, c-format
+msgid "failed to bind key %d to value %d\n"
+msgstr "przypisanie klawisza %d do warto¶ci %d nie powiod³o siê\n"
+
+#: loadkeys.y:817
+#, c-format
+msgid "deallocate keymap %d\n"
+msgstr "zwalnianie mapy klawiszy %d\n"
+
+#: loadkeys.y:823
+#, c-format
+msgid "%s: could not deallocate keymap %d\n"
+msgstr "%s: nie mo¿na zwolniæ mapy %d\n"
+
+#: loadkeys.y:838
+#, c-format
+msgid "%s: cannot deallocate or clear keymap\n"
+msgstr "%s: nie mo¿na zwolniæ ani wyczy¶ciæ mapy\n"
+
+#: loadkeys.y:849
+#, c-format
+msgid "%s: failed to restore keyboard mode\n"
+msgstr "%s: przywrócenie trybu klawiatury nie powiod³o siê\n"
+
+#: loadkeys.y:852
+#, c-format
+msgid ""
+"%s: warning: this map uses Unicode symbols\n"
+"    (perhaps you want to do `kbd_mode -u'?)\n"
+msgstr ""
+"%s: uwaga: ta mapa u¿ywa symboli unikodowych\n"
+"    (zapewne trzeba jeszcze wywo³aæ `kbd_mode -u'?)\n"
+
+#: loadkeys.y:896
+#, c-format
+msgid "failed to bind string '%s' to function %s\n"
+msgstr "przypisanie ³añcucha '%s' do funkcji %s nie powiod³o siê\n"
+
+#: loadkeys.y:903
+#, c-format
+msgid "failed to clear string %s\n"
+msgstr "czyszczenie ³añcucha %s nie powiod³o siê\n"
+
+#: loadkeys.y:920
+#, c-format
+msgid "too many compose definitions\n"
+msgstr "za du¿o definicji compose\n"
+
+#: loadkeys.y:980
+msgid "impossible error in do_constant"
+msgstr "niemo¿liwy b³±d w do_constant"
+
+#: loadkeys.y:998
+#, c-format
+msgid ""
+"\n"
+"Changed %d %s and %d %s.\n"
+msgstr ""
+"\n"
+"Zmieniono %d %s i %d %s.\n"
+
+#: loadkeys.y:999
+msgid "key"
+msgstr "klawisz"
+
+#: loadkeys.y:999
+msgid "keys"
+msgstr "klawiszy"
+
+#: loadkeys.y:1000
+msgid "string"
+msgstr "³añcuch"
+
+#: loadkeys.y:1000
+msgid "strings"
+msgstr "³añcuchów"
+
+#: loadkeys.y:1002
+#, c-format
+msgid "Loaded %d compose %s.\n"
+msgstr "Wczytano %d %s compose.\n"
+
+#: loadkeys.y:1003
+msgid "definition"
+msgstr "definicjê"
+
+#: loadkeys.y:1003
+msgid "definitions"
+msgstr "definicji"
+
+#: loadkeys.y:1005
+#, c-format
+msgid "(No change in compose definitions.)\n"
+msgstr "(Brak zmian w definicjach compose.)\n"
+
+#: loadkeys.y:1038
+#, c-format
+msgid "loadkeys: don't know how to compose for %s\n"
+msgstr "loadkeys: nie wiadomo jak zrobiæ compose dla %s\n"
+
 #: src/loadunimap.c:42
 #, c-format
 msgid ""
