--- kbd-1.05/data/keymaps/i386/include/compose.inc.compose	Wed May  2 18:53:30 2001
+++ kbd-1.05/data/keymaps/i386/include/compose.inc	Wed May  2 18:54:29 2001
@@ -0,0 +1,8 @@
+# We map the PrintScreen key (useless) and Alt+AltGr to the Compose key...
+# A really useful thing not found on normal keyboards. :/
+#
+# (c) 2001 Bernhard Rosenkränzer <bero@redhat.com>
+
+keycode  99 = Compose
+keycode 100 = AltGr
+	alt	keycode 100 = Compose
--- kbd-1.05/data/keymaps/i386/include/qwerty-layout.inc.compose	Wed May  2 18:55:06 2001
+++ kbd-1.05/data/keymaps/i386/include/qwerty-layout.inc	Wed May  2 18:56:12 2001
@@ -28,3 +28,4 @@
 keycode 49 = n
 keycode 50 = m
 #
+include "compose"
--- kbd-1.05/data/keymaps/i386/include/qwertz-layout.inc.compose	Wed May  2 18:55:13 2001
+++ kbd-1.05/data/keymaps/i386/include/qwertz-layout.inc	Wed May  2 18:55:23 2001
@@ -28,3 +28,4 @@
 keycode 49 = n
 keycode 50 = m
 #
+include "compose"
